docs/csharp/query-expression-basics.md

```diff
@@ -0,0 +1,179 @@
+---
+title: "Query expression basics"
+description: Introduces concepts related to query expressions
+keywords: .NET, .NET Core, C#
+author: stevehoag
+manager: wpickett
+ms.author: wiwagn
+ms.date: 11/30/2016
+ms.topic: article
+ms.prod: .net-core
+ms.technology: .net-core-technologies
+ms.devlang: dotnet
+ms.assetid: 027db1f8-346f-44d2-a16e-043fcea3a4e0
+---
+# Query expression basics
+
+## What is a query and what does it do? 
+
+ A *query* is a set of instructions that describes what data to retrieve from a given data source (or sources) and what shape and organization the returned data should have. A query is distinct from the results that it produces.  
+  
+ Generally, the source data is organized logically as a sequence of elements of the same kind. For example, a SQL database table contains a sequence of rows. In an XML file, there is a "sequence" of XML elements (although these are organized hierarchically in a tree structure). An in-memory collection contains a sequence of objects. 
+  
+ From an application's viewpoint, the specific type and structure of the original source data is not important. The application always sees the source data as an <xref:System.Collections.Generic.IEnumerable%601> or <xref:System.Linq.IQueryable%601> collection. For example, in LINQ to XML, the source data is made visible as an `IEnumerable`\<<xref:System.Xml.Linq.XElement>>.  
+  
+ Given this source sequence, a query may do one of three things:  
+  
+-   Retrieve a subset of the elements to produce a new sequence without modifying the individual elements. The query may then sort or group the returned sequence in various ways, as shown in the following example (assume `scores` is an `int[]`):  
+  
+     [!code-cs[csrefQueryExpBasics#45](../../samples/snippets/csharp/concepts/linq/query-expression-basics_1.cs)]  
+  
+-   Retrieve a sequence of elements as in the previous example but transform them to a new type of object. For example, a query may retrieve only the last names from certain customer records in a data source. Or it may retrieve the complete record and then use it to construct another in-memory object type or even XML data before generating the final result sequence. The following example shows a transform from an `int` to a `string`. Note the new type of `highScoresQuery`.  
```

Comment:
Fixed - no idea where the term "transform" came from. This was written a long time ago, so maybe terminology has changed over the years?